tools/rosgraph/CMakeLists.txt

```diff
@@ -17,4 +17,5 @@ catkin_install_python(PROGRAMS scripts/rosgraph
 
 if(CATKIN_ENABLE_TESTING)
   catkin_add_nosetests(test)
+  catkin_add_nosetests(test/isolated/test_roslogging_user_logger.py)
```

Comment:
I haven't tried it but the problem might be related to the fact that you do things in the module level of the test file. Please try to move all logic into a test function (and eventually a setup and teardown function).